src/sentry/utils/snowflake.py

```diff
@@ -73,62 +58,27 @@ def msb_0_ordering(value, width):
     return int(msb_0_ordering, 2)
 
 
-def generate_snowflake_id(redis_key: str) -> int:
-    segment_values = {}
+def generate_snowflake_id(redis_key: str = "") -> int:
+    from sentry.models import SnowflakeSeq
 
+    segment_values = {}
     segment_values[VERSION_ID] = msb_0_ordering(settings.SNOWFLAKE_VERSION_ID, VERSION_ID.length)
 
     try:
         segment_values[REGION_ID] = get_local_region().id
     except RegionContextError:  # expected if running in monolith mode
         segment_values[REGION_ID] = NULL_REGION_ID
 
-    current_time = datetime.now().timestamp()
+    sequence = SnowflakeSeq.next_seq()
     # supports up to 130 years
-    segment_values[TIME_DIFFERENCE] = int(current_time - settings.SENTRY_SNOWFLAKE_EPOCH_START)
```

Comment:
Without a lot more discussion, I'd prefer not to update the version, maintain the format as defined, and just update the sequence implementation